### PR TITLE
WC2-943: Display Storage IDs as HEX

### PR DESCRIPTION
**Pull request by bmonjoie**
## What problem is this PR solving?

Storage IDs are displayed as HEX on some cards; we should therefore be able to match what is displayed on the card.

### Related JIRA tickets

WC2-943

## Changes

Add the HEX value in front of the base64 value

## How to test

Have some Storage device with a BASE64 customer chosen ID and go in the External Storage screen.

Make sure to try to search for the HEX string as well.

## Print screen / video

image

## Notes

Example values are given in the ticket.

### CODE REVIEW

File: iaso/api/storage.py [hunk 1]
```diff
@@ -133,6 +136,14 @@ class Meta:
             "entity",
         )
 
+    @staticmethod
+    def get_storage_id(obj: StorageDevice):
+        base64_id = obj.customer_chosen_id
+        try:
+            return f"{base64.b64decode(base64_id).hex(' ').upper()} ({base64_id})"
+        except (TypeError, binascii.Error):
+            return base64_id
```
Comment (hugo093) [inline iaso/api/storage.py:145]: why don't you reuse the annotation in your filter (and pass it to the queryset in the viewset) ?  

You could even put that annotation as a queryset method. 

That way you could also easily test the fact that hex_id should return None in case of error (aka not a base64)

**Review by hugo093 — CHANGES_REQUESTED** [review]

File: iaso/api/storage.py (hunk 1 above)
Comment (bmonjoie) [reply iaso/api/storage.py:145]: Because the annotation is not formatted the way we want (a space every 2 characters).

Comment (hugo093) [reply iaso/api/storage.py:145]: then lgtm

**Review by hugo093 — APPROVED** [review]
